src/zenml/pipelines/base_pipeline.py

```diff
@@ -660,11 +660,32 @@ def run(
             finally:
                 constants.SHOULD_PREVENT_PIPELINE_EXECUTION = False
 
-            # Log the dashboard URL
-            dashboard_utils.print_run_url(
-                run_name=deployment.run_name_template,
-                pipeline_id=pipeline_id,
-            )
+            if (
+                deployment_model
+                and deployment_model.pipeline
+                and deployment_model.stack
+            ):
+                runs = Client().list_runs(
```

Comment:
I'm guessing since the deployment has a unique UUID, we could make this simpler and just filter by deployment ID (skipping the stack/workspace/pipeline)?
There won't be any functional difference to the current solution however, so I'll leave that up to you.

Reply:
Yeah that was my thought going in, but you did mention that we'd want to filter by other params as well so I thought maybe you were thinking of something that I wasn't. I think it's cleaner without the stack / workspace + pipeline tbh.

Reply:
Ah yes good point. `desc:created` does what we want it to (i.e. gets the most recent run)

Reply:
Not sure if it's relevant (probably there is no possibility that multiple runs exist at that point), but we actually want the first one to get the run that was triggered by the `pipeline.run()` call.

Reply:
AFAICS what the code now does get that first one. Actually it should get that just by specifying the `deployment_model.id`, unless I'm missing something?